### ISSUE TITLE
[no-untyped-public-signature] Throws on constructor, when a constructor can't have a return type

### ISSUE THREAD

**Issue by garyking** [typescript-eslint/typescript-eslint#1229]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-untyped-public-signature": "error"
  }
}
```

```TS
export default class Foo {
  private readonly urlPath: string;

  /**
   * Public method has no return type
   *
   * eslint(@typescript-eslint/no-untyped-public-signature)
   */
  public constructor(urlPath: string) {
    this.urlPath = urlPath;
  }
}
```

**Expected Result**

No error, because constructor can't have return type.

**Actual Result**

The error as shown in the comments.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.8.0` |
| `@typescript-eslint/parser`        | `2.8.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `v12.12.0` |
| `npm`                              | `6.13.1` |
